File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/dao/UserDao.kt
package dev.tpavani.octopoints.db.dao

import androidx.room.Dao
import androidx.room.Query
import dev.tpavani.octopoints.db.entity.UserEntity

@Dao
interface UserDao : IDao<UserEntity> {
    @Query("SELECT * FROM users ORDER BY username")
    suspend fun getAllUsers() : List<UserEntity>
}

File: octopoints_flutter/lib/db/dao/UserDao.dart
import 'package:floor_annotation/floor_annotation.dart';
import 'package:octopoints_flutter/db/dao/IDao.dart';
import 'package:octopoints_flutter/db/entity/UserEntity.dart';

@dao
abstract class UserDao extends IDao<UserEntity> {
  @Query('SELECT * FROM users ORDER BY username')
  Future<List<UserEntity>> getAllUsers();

  @Query('SELECT * FROM users WHERE id=:id')
  Future<UserEntity?> getUserById(int id);
}
